math/binary_exponentiation.cpp

```diff
@@ -0,0 +1,58 @@
+#include<bits/stdc++.h>
```

Comment:
This is discouraged. Please use each header/library you need instead of this global header/library.